src/Workspaces/Core/Portable/Workspace/Workspace_Editor.cs

```diff
@@ -636,13 +711,10 @@ protected virtual Solution AdjustReloadedSolution(Solution oldSolution, Solution
             // keep open documents using same text
             foreach (var docId in this.GetOpenDocumentIds())
             {
-                if (newSolution.ContainsDocument(docId))
+                var document = newSolution.GetTextDocument(docId);
```

Comment:
so, now GetOpenDocumentIds can return id that belong to analyzer config, shouldn't we need to go through all caller and make sure they don't expect id to be only either document or additional document?

Reply:
I discussed this a bunch with @jasonmalinowski last time we switched `GetOpenDocumentIds` from returning just Documents to returning both Documents and AdditionalDocuments, and yes we figured it is a functionality breaking change, but most likely a good one, where fixing the caller would be the right thing. When I did that change, I did audit all the callsites in Roslyn.sln and fixed those that made the assumption about returned document kind.

Reply:
We got somewhat lucky: the Ids that are open in this case are associated with a Workspace -- anybody who actually wanted the document had to call Solution.GetDocument(), which conveniently returns null. Since you had to do that *anyways* to deal with the race of a document being removed between the call to GetOpenDocumentIds and the fetching of the document, it was fine.